js/main.js

```diff
@@ -1,45 +1,491 @@
+    PLAYER_BOARD[rows][columns] = 2;
+  }
+  if (PLAYER_BOARD[rows][columns] === 0) {
+    randomDiv.classList.add("miss");
+    PLAYER_BOARD[rows][columns] = 3;
+  }
+  turn = 1;
+  activeTurn();
+}
+
+//Finding if there is an index around the last hit div that can be fired at again
+function getSuitableIndex(board) {
+  if (
+    lastHitColumn + 1 !== 10 &&
+    board[lastHitRow][lastHitColumn + 1] !== 2 &&
+    board[lastHitRow][lastHitColumn + 1] !== 3
+  ) {
+    return `${lastHitRow}${lastHitColumn + 1}`;
+  } else if (
+    lastHitRow + 1 !== 10 &&
+    board[lastHitRow + 1][lastHitColumn] !== 2 &&
+    board[lastHitRow + 1][lastHitColumn] !== 3
+  ) {
+    return `${lastHitRow + 1}${lastHitColumn}`;
+  } else if (
+    lastHitColumn - 1 !== -1 &&
+    board[lastHitRow][lastHitColumn - 1] !== 2 &&
+    board[lastHitRow][lastHitColumn - 1] !== 3
+  ) {
+    return `${lastHitRow}${lastHitColumn - 1}`;
+  } else if (
+    lastHitRow - 1 !== -1 &&
+    board[lastHitRow - 1][lastHitColumn] !== 2 &&
+    board[lastHitRow - 1][lastHitColumn] !== 3
+  ) {
+    return `${lastHitRow - 1}${lastHitColumn}`;
+  } else return null;
+}
+
+//Get the correct ship object from the clicked div
+function findCorrectShip(board, div) {
+  let correctShip;
+  Object.values(board === PLAYER_BOARD ? playerShips : cpuShips).forEach(
+    (obj) => {
+      if (obj.key === div.id) correctShip = obj;
+    }
+  );
+  return correctShip;
+}
+
+//Checking if the ship that was hit has been sunk
+function checkIfSunk(board, ship, div) {
+  let currentPlayer = board === PLAYER_BOARD ? PLAYER_DIVS : CPU_DIVS;
+  if (ship.health === 0 && [...div.classList].includes("horizontal")) {
+    for (let i = 0; i < ship.length; i++) {
+      currentPlayer[parseInt(ship.firstIndex) + i].classList.add("sunken-ship");
+    }
+    if (board === PLAYER_BOARD) {
+      suitableIndex = null;
+      lastHitRow = null;
+      lastHitColumn = null;
+    }
+  }
+  if (ship.health === 0 && [...div.classList].includes("vertical")) {
+    for (let i = 0; i < ship.length * 10; i += 10) {
+      currentPlayer[parseInt(ship.firstIndex) + i].classList.add("sunken-ship");
+    }
+    if (board === PLAYER_BOARD) {
+      suitableIndex = null;
+      lastHitRow = null;
+      lastHitColumn = null;
+    }
+  }
+  getWinner(board);
+}
+//Checking if a winner has been found
+function getWinner(board) {
+  let sunkShipsCounter = 0;
+  let currentPlayer = board == PLAYER_BOARD ? playerShips : cpuShips;
+  Object.values(currentPlayer).forEach((obj) => {
+    obj.health === 0 ? sunkShipsCounter++ : sunkShipsCounter;
+  });
+  if (sunkShipsCounter === Object.keys(currentPlayer).length) {
+    winner = currentPlayer === cpuShips ? 1 : -1;
+    activeTurn();
+  }
+}
+
+//Removing the existing divs
+function resetBoard(board, divs) {
+  board.forEach((arrEl) => {
+    arrEl.forEach((el, innerIdx) => {
+      arrEl[innerIdx] = 0;
+    });
+  });
+  divs.forEach((div) => {
```

Comment:
If you're just trying to reset `divs` into an empty array, you can just assign a new empty array to it. `divs = []`

Reply:
I tried this while refactoring and it was causing my game to bug out and not removing the existing divs. Perhaps because the divs are both in constants? Not throwing any console error about assigning to a constant just duplicating divs.

Reply:
Ah yeah, likely because you're referencing those 2 variables in other variables in your code. In that case you should be able to just do `divs.length = 0`. No need to do the looped `remove()`